machine/v1beta1/types_awsprovider.go

```diff
@@ -84,6 +84,13 @@ type AWSMachineProviderConfig struct {
 	// When omitted, no placement group is used when creating the EC2 instance.
 	// +optional
 	PlacementGroupName string `json:"placementGroupName,omitempty"`
+	// PlacementGroupPartition is the partition number within the placement group in which to launch the instance.
+	// This value is only valid if the placement group, referred in `PlacementGroupName`, was created with
+	// strategy set to partition.
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=7
+	// +optional
+	PlacementGroupPartition int64 `json:"placementGroupPartition,omitempty"`
```

Comment:
Why int64? Change to int32, we only need to support small numbers here

Reply:
The aws-sdk uses pointer to int64 to specify the partition number. 
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.50.0/service/ec2#Placement.PartitionNumber

xref:
```go
	PartitionNumber *int64 `locationName:"partitionNumber" type:"integer"`
```

In order to match with it and [CAPA PR](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4883), I chose int64 here as well.

Reply:
In general we prefer to build APIs that make sense for us, rather than being restricted by cloud provider APIs. But given the context of this and the fact that we are going to convert to CAPI in the near future, there's value in this being the same.

Do you think there's any likelihood of being able to change the upstream to int32? Since that would make more sense for the Kube API

Reply:
Updating it to int32 would be better, making more sense to our API. We can anyway convert it to int64 in the provider code to match with aws-sdk. 

For upstream, I'll try to open a PR to fix it.